pkg/loop/telem.go

```diff
@@ -58,13 +61,24 @@ func SetupTracing(config TracingConfig) error {
 	}
 
 	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
 
-	// Set up a trace exporter
-	// Shutting down the traceExporter will not shutdown the underlying connection.
-	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithEndpoint(config.CollectorTarget), otlptracegrpc.WithDialOption(
+	conn, err := grpc.DialContext(ctx, config.CollectorTarget,
 		// Note the use of insecure transport here. TLS is recommended in production.
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
-	))
+		grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+			conn, err := net.Dial("tcp", s)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to dial: %v", err)
```

Comment:
Something straightforward seems simplest to me :shrug: 
```suggestion
				config.OnDialError(err)
```
```diff
type TracingConfig struct {
	// NodeAttributes are the attributes to attach to traces.
	NodeAttributes map[string]string
	// Enables tracing; requires a collector to be provided
	Enabled bool
	// Collector is the address of the OTEL collector to send traces to.
	CollectorTarget string
	// SamplingRatio is the ratio of traces to sample. 1.0 means sample all traces.
	SamplingRatio float64
+	// OnDialError is called when the dialer fails, providing an opportunity to log.
+	OnDialError func(error)
}
```
```go
config.OnDialError = func(err error) { lggr.Errorw("Failed to dial", "err", err) }
```